cypress/pages/employeePage.ts

```diff
@@ -0,0 +1,158 @@
+    deleteUserApiUrl = "https://opensource-demo.orangehrmlive.com/web/index.php/api/v2/pim/employees"
+    searchResultMainUrl = "https://opensource-demo.orangehrmlive.com/web/index.php/api/v2/pim/employees?limit=50&offset=0&model=detailed&nameOrId="
+    allEmployeeApi = "https://opensource-demo.orangehrmlive.com/web/index.php/api/v2/pim/employees"
+    searchResultEndUrl = "&includeEmployees=onlyCurrent&sortField=employee.firstName&sortOrder=ASC"
+    personalDetailsApi = "https://opensource-demo.orangehrmlive.com/web/index.php/api/v2/pim/employees/"
+
+
+
+    setUserData(firstName, middleName, lastName, userId, userName, userStatus, password) {
+
+        const setValue = (setSetter: string[], setValue: string[]) => {
+
+            for (let elem in setSetter) {
+
+                cy.task("UserDataSetter", { setter: setSetter[elem], value: setValue[elem] })
+            }
+
+        }
+        const setterArray = ["firstName", "middleName", "lastName", "userId", "userName", "userStatus", "password"]
+        const setValueArray = [firstName, middleName, lastName, userId, userName,userStatus, password]
+
+        setValue(setterArray, setValueArray)
+
+    }
+
+    /**
+       * Add current employee id and name to url of api.
+       *
+       * @param {number} id - The id name of the employee.
+       * @param {string} name - The first name of the employee.
+       *
+       */
+    specifyIdinURL(id: number, name: string) {
+
+        return `${this.searchResultMainUrl}${name}&employeeId=${id}${this.searchResultEndUrl}`
+    }
+
+    /**
+     * Fill in the employee details fields with the specified information.
+     *
+     * @param {string} firstName - The first name of the employee.
+     * @param {string} middleName - The middle name of the employee.
+     * @param {string} lastName - The last name of the employee.
+     * @param {string} userId - The user ID of the employee.
+     * @param {string} userName - The user name of the employee.
+     * @param {status} userStatus - The status of the user (e.g., 'Enabled', 'Disabled').
+     *
+     */
+    fillInEmployeeDetailsFields(firstName: string, middleName: string, lastName: string, userId: string, userName: string, userStatus: string) {
+        const getAndType = (selector: string, value: string) => {
+
+            cy.get(selector).should("have.text", "")
+            cy.get(selector).type(value);
+        }
+
+        const getAndAssertToBeEmpty = (selector: string, index: number) => {
+            cy.get(selector).eq(index).clear().invoke("val").then((value) => {
+                expect(value).to.be.empty
+            })
+        }
+
+        getAndType(this.firstNameField, firstName)
+        getAndType(this.middleNameField, middleName)
+        getAndType(this.lastNameField, lastName)
+
+        cy.get(this.userIDField).eq(4).clear().invoke("val").then((value) => {
+            expect(value).to.be.empty
+        })
+
+        cy.get(this.userIDField).eq(4).type(userId);
```

Comment:
since we are using userIDField selector with .eq(4)  always, we can update the selector to get the 4th element directly